examples/step-57/step-57.cc

```diff
@@ -0,0 +1,879 @@
+                      << "        Do refinement ------   " << std::endl;
+          }
+      }
+
+
+  }
+
+  // @sect4{StationaryNavierStokes::refine_mesh}
+  //
+  // After finding a good initial guess on the coarse mesh, we hope to
+  // decrease the error through refining the mesh. Here we do adaptive
+  // refinement similar to step-15 except that we use the Kelly estimator on
+  // the velocity only. We also need to transfer the current solution to the
+  // next mesh using the SolutionTransfer class.
+  template <int dim>
+  void StationaryNavierStokes<dim>::refine_mesh()
+  {
+    Vector<float> estimated_error_per_cell (triangulation.n_active_cells());
+    FEValuesExtractors::Vector velocity(0);
+    KellyErrorEstimator<dim>::estimate (dof_handler,
+                                        QGauss<dim-1>(degree+1),
+                                        typename FunctionMap<dim>::type(),
+                                        present_solution,
+                                        estimated_error_per_cell,
+                                        fe.component_mask(velocity));
+
+    GridRefinement::refine_and_coarsen_fixed_number (triangulation,
+                                                     estimated_error_per_cell,
+                                                     0.3, 0.0);
+
+    triangulation.prepare_coarsening_and_refinement();
+    SolutionTransfer<dim, BlockVector<double> > solution_transfer(dof_handler);
+    solution_transfer.prepare_for_coarsening_and_refinement(present_solution);
+    triangulation.execute_coarsening_and_refinement ();
+
+    // First the DoFHandler is set up and constraints are generated. Then we
+    // create a temporary vector "tmp", whose size is according with the
+    // solution on the new mesh.
+    setup_dof();
+
+    BlockVector<double> tmp (dofs_per_block);
+
+    // Transfer solution from coarse to fine mesh and apply boundary value
+    // constraints to the new transfered solution. Note that present_solution
+    // is still a vector corresponding to the old mesh.
+    solution_transfer.interpolate(present_solution, tmp);
+    nonzero_constraints.distribute(tmp);
+
+    // Finally set up matrix and vectors and set the present_solution to the
+    // interpolated data.
+    initialize_system();
+    present_solution = tmp;
+  }
+
+  // @sect4{StationaryNavierStokes::search_initial_guess}
+  //
+  // This function will provide us with an initial guess by using a
+  // continuation method as we discussed in the introduction. The Reynolds
+  // number is increased step-by-step until we reach the target value. By
+  // experiment, the solution to Stokes is good enough to be the initial guess
+  // of NSE with Reynolds number 1000 so we start there.  To make sure the
+  // solution from previous problem is close enough to the next one, the step
+  // size must be small enough.
+  template <int dim>
+  void StationaryNavierStokes<dim>::search_initial_guess(double step_size)
```

Comment:
In computer science, the word "search" is usually used to look for an element in a finite and existing collection that matches a condition. I think `find_initial_guess()` or `compute_initial_guess()` would be a better choice of name for the function.

As for the previous one, it may be worth moving it higher up in the file.